flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -130,7 +131,9 @@ private ScanContext(
     this.watermarkColumn = watermarkColumn;
     this.watermarkColumnTimeUnit = watermarkColumnTimeUnit;
 
-    validate();
+    if (!skipValidate) {
```

Comment:
is this only for the endSnapshotId? if yes, should we remove the validation on endSnapshotId?

Reply:
> If yes, then we change the copyWithAppendsBetween and the  copyWithSnapshotId to remove the streaming flag, as those are not a streaming scans anymore.

that is also not correct, because `copy` should meant `copy`. The main problem is that `ScanContext` is used for both user intention (via source builder) and internal incremental scan. I agree that internal incremental scan shouldn't have the streaming setting. 

note that `ScanContext` is not a public class. Users can't construct this object directly. Maybe the `validate()` method shouldn't be called by the constructor and only be called by the `ScanContext#Builder#build()` method?

Reply:
Went with your suggestion, and removed the generic validation, and called the validation explicitly from the source.